### PR TITLE
Issue112: Profile Page API (information update to DB not implemented yet)

### PR DESCRIPTION
**Pull request by Andrii-I**

### CODE REVIEW

**Comment by Andrii-I** [comment]
closes #126
closes #112

File: src/client/pages/Profile.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+          <div className="col-left">
+            <h3 className="profile-modal-section-header">Profile picture</h3>
+            <img src="/src/client/style/images/sample_profile_pic.png" alt="profile pic" />
+            <br />
+            <a href="#profilePicEdit">Edit</a>
+            {' '}
+            |
+            {' '}
+            <a href="#profilePicRemove">Remove</a>
+            <br />
```
Comment (zzheng07) [inline src/client/pages/Profile.js:111]: It's my understanding that these `br` tags are used for spacing. If that's the case, I consider them bad smell in the code, making the code base cluttered. I suggest resolving them in near future.

File: src/server/app.js [hunk 2]
```diff
@@ -18,6 +18,27 @@ app.get('/api/disorder/search', (req, res) => {
   });
 });
 
+app.get('/api/getProfile/:ID', (req, res) => {
```
Comment (zzheng07) [inline src/server/app.js:21]: I suggest using one endpoint for profile api. `api.get('/api/profile...')` and ``api.put('/api/profile...')``

File: src/client/pages/Profile.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  const [about, setAbout] = useState();
+  const [name, setName] = useState();
+  const [loc, setLocation] = useState();
+  const [verified, setVerified] = useState();
+  const [DOB, setDOB] = useState();
+  const [ID, setID] = useState(2);
+
+  const endPoint = `/api/getProfile/${ID}`;
+  const [isLoading, data, error] = useFetch(endPoint);
+  const dataList = data || [];
```
Comment (zzheng07) [inline src/client/pages/Profile.js:24]: The naming here does not make sense. Is it a list or a single object?

File: src/client/pages/Profile.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+
+  useEffect(() => {
+    setUsername(dataList.Username);
```
Comment (zzheng07) [inline src/client/pages/Profile.js:28]: Please use lower case for object properties. Those may be from the backend, then looking into backend.

File: src/client/pages/Profile.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    const request = {
+      method: 'PUT',
+      mode: 'cors',
+      headers: {
+        'Content-Type': 'application/json',
+        'Content-Length': profileData.toString().length.toString()
+      },
+      body: JSON.stringify(profileData)
+    };
+    fetch(updatePoint, request);
```
Comment (zzheng07) [inline src/client/pages/Profile.js:66]: If it is an asynchronous call here, don't forget to handle callback.

File: src/client/pages/Profile.js (hunk 1 above)
Comment (Andrii-I) [reply src/client/pages/Profile.js:111]: Ok.
Yes, they are used for layout formatting. What is a better alternative?

File: src/server/app.js (hunk 2 above)
Comment (Andrii-I) [reply src/server/app.js:21]: Ok. Get API is still work in progress so I will incorporate the comment in next update

File: src/client/pages/Profile.js (hunk 5 above)
Comment (Andrii-I) [reply src/client/pages/Profile.js:66]: Ok. Get API is still work in progress so I will incorporate the comment in next update

File: src/client/pages/Profile.js (hunk 3 above)
Comment (Andrii-I) [reply src/client/pages/Profile.js:24]: I have a number of fields in the user profile and in the DB  (columns of a single row of a table), each field is a constant here, dataList is either data from API or empty data. Taking this into account, how can I improve the implementation?

File: src/client/pages/Profile.js (hunk 4 above)
Comment (Andrii-I) [reply src/client/pages/Profile.js:28]: which part of the code is object property? Is it "setUsername" or "dataList" or "Username" or all of them?

**Pull request merged**